.github/workflows/alpine-unittests.yml

```diff
@@ -0,0 +1,68 @@
+
+name: Alpine Unittests
+
+on:
+  pull_request:
+    branches-ignore:
+      - 'ubuntu/**'
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: 'ci-${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: sh -ex {0}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: "Checkout"
+        uses: actions/checkout@v4
+        with:
+          # Fetch all tags for tools/read-version
+          fetch-depth: 0
+
+      - name: Install incus
+        run: curl https://pkgs.zabbly.com/get/incus-stable | sudo sh -x
+
+      - name: Reset the firewall
+        run: |
+          sudo iptables -I DOCKER-USER -j ACCEPT
+
+      - name: Init incus
+        run: |
+          sudo incus admin init --auto
+
+          # we shouldn't have to do this, but the link is down and we need it up
+          sudo ip link set incusbr0 up
+
+      - name: Create alpine container
+        # the current shell doesn't have incus-admin as one of the groups
+        # so switch groups to run incus commands
+        run: sudo sg incus-admin -c 'incus launch images:alpine/edge alpine'
+
+      - name: Check networking (for debugging)
+        run: |
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 google.com || true'
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 dl-cdn.alpinelinux.org || true'
+          sudo sg incus-admin -c 'incus exec alpine -- nslookup www.google.com || true'
+          sudo sg incus-admin -c 'incus exec alpine -- ping -c 1 dl-cdn.alpinelinux.org || true'
+
+      - name: Install dependencies
+        run: sudo sg incus-admin -c 'incus exec alpine -- apk add py3-tox git'
```

Comment:
> https://github.com/canonical/cloud-init/actions/runs/8668134702/job/23775704648?pr=5121

I could not reproduce the error with the latest image. This sounds to me like an inconsistent state of the repository indexes within the image. Likely avoidable by prepending an `apk update` and optionally an `apk upgrade` before installing `py3-tox`.